x/qgb/keeper/msg_server.go

```diff
@@ -142,6 +153,28 @@ func (k msgServer) DataCommitmentConfirm(
 		return nil, sdkerrors.Wrap(types.ErrInvalid, "submitted eth address does not match delegate eth address")
 	}
 
+	// Verify if validator is part of the previous valset
+	var previousValset *types.Valset
+	// TODO add test for case nonce == 1.
+	if msg.Nonce == 1 {
+		// if the msg.Nonce == 1, the current valset should sign the first valset
```

Comment:
```suggestion
		// if the msg.Nonce == 1, the current valset should sign the first valset. Because, it's the first attestation, and there is no prior validator set defined that should sign this change.
```